src/pvrclient-mediaportal.cpp

```diff
@@ -2481,3 +2482,52 @@ PVR_ERROR cPVRClientMediaPortal::GetStreamReadChunkSize(int& chunksize)
   chunksize = 32 * 1024;
   return PVR_ERROR_NO_ERROR;
 }
+PVR_ERROR cPVRClientMediaPortal::GetRecordingEdl(const kodi::addon::PVRRecording& recording, std::vector<kodi::addon::PVREDLEntry>& edl) // bilkusg
+// liberally adapted from pvr.vuplus ( recordings.cpp in particular)
+{
+    cRecording* myrecording = GetRecordingInfo(recording);
+    const char* pvrFilePath = myrecording->FilePath();
+    const std::string& pvrSmbPath = m_tsreader->TranslatePath(pvrFilePath);
+    const std::string edlFile = pvrSmbPath.substr(0, pvrSmbPath.find_last_of('.')) + ".edl";
+    kodi::Log(ADDON_LOG_INFO, "GetRecordingEdl for %s from %s", pvrFilePath,edlFile.c_str());
+    FileReader *fileReader = new FileReader();
+    long fileOpenResult = fileReader->OpenFile(edlFile);
+    size_t fileSize = fileReader->GetFileSize();
+    unsigned char fileContentsChars[10000];//  = new unsigned char(fileSize + 1); // edl files aren't that large, so this is reasonable
+    size_t bytesRead = 0;
+    fileReader->Read(fileContentsChars, fileSize, &bytesRead);
```

Comment:
I see a hard-coded buffer size of 10000 but what if fileSize is larger?
Maybe better to allocate a dynamic buffer using the fileSize using a new[] + corresponding delete[]

Reply:
I think this is superseded below, but I did it this way because using new and delete caused a malloc error, and I saw there's lots of other stuff in there about dynamic allocating data. 10k is pretty large for an edl file I think, but if the direct use of kodi's file class will meet the requirement better, I'll do it that way.